components/datetime/src/neo.rs

```diff
@@ -453,8 +454,10 @@ size_test!(
 #[doc = neo_year_month_day_formatter_size!()]
 #[derive(Debug, Clone)]
 pub struct DateTimeFormatter<FSet: DateTimeNamesMarker> {
-    selection: DateTimeZonePatternSelectionData,
+    // Store the kind in here for better niche optimization.
+    pub(crate) selection: DateTimeZonePatternSelectionData<AnyCalendarForFormattingKind>,
```

Comment:
I had to do this in c9d32d77a6314fca974748dfed4381410ee7bad3 because the new field pushed the struct up to be an extra word in size.

I could alternatively not store the `AnyCalendarForFormattingKind`, since it is only needed in the niche (no pun intended) use case of converting it to a DateTimeNames. I can just re-compute it at that point, with a debug assertion for the unreachable case.

Reply:
Ok I reverted that commit and instead pushed 843b8ab55c57c46ae87e471bf709490a08c72bb4 which just makes an untagged struct.